src/controllers/localWebService.ts

```diff
@@ -57,16 +82,38 @@ export default class LocalWebService {
     }
 
     broadcast(uri: string, event: string, data?: any): void {
-        let temp = {
-            type: event
+        // Create a message to send out
+        let message = {
+            type: event,
+            data: data ? data : undefined
         };
 
-        if (data) {
-            temp['data'] = data;
+        // Attempt to find the web socket server
+        let mapping = this.wsMap.get(uri);
+
+        // Is the URI mapped to a web socket server?
+        if (mapping === undefined) {
+            // There isn't a mapping, so create it
+            mapping = new WebSocketMapping();
+            this.wsMap.set(uri, mapping);
+        } else {
+            // Make sure the web socket server is open, then fire away
+            if (mapping.webSocketServer.readyState === ws.OPEN) {
+                mapping.webSocketServer.send(JSON.stringify(message));
+            }
         }
 
-        if (this.clientMap.has(uri) && this.clientMap.get(uri).readyState === ws.OPEN) {
-            this.clientMap.get(uri).send(JSON.stringify(temp));
+        // Append the message to the message history
+        mapping.pendingMessages.push(message);
+    }
+
+    /**
+     * Purges the queue of messages to send on the web socket server for the given uri
+     * @param   uri URI of the web socket server to reset
+     */
+    resetSocket(uri: string): void {
+        if (this.wsMap.has(uri)) {
+            this.wsMap.delete(uri);
```

Comment:
Theres a better user experience that could happen here, but it's not really necessary atm.